tests/scenario/test_deploy.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from unittest.mock import Mock, PropertyMock
+
+from charms.harness_extensions.v0.evt_sequences import Event, Scenario
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
+from ops.model import BlockedStatus, Container, WaitingStatus
+from ops.pebble import Error as PebbleError
+from ops.pebble import PathError
+
+from charm import MimirK8SOperatorCharm
+
+
+def setup_function():
+    Container.can_connect = Mock(return_value=True)
+    MimirK8SOperatorCharm._mimir_version = PropertyMock(return_value="2.4.0")
+    MimirK8SOperatorCharm._current_mimir_config = PropertyMock(return_value={})
+    MimirK8SOperatorCharm._set_alerts = Mock(return_value=True)
+    KubernetesServicePatch.__init__ = Mock(return_value=None)
+
+
+def generate_scenario():
```

Comment:
This function is doing more than generating a scenario: it's actually playing (i.e. simulating) a bunch of events with a null context.
I'd call this `test_startup_sequence`.

Reply:
Yes, you are right it's actually playing a bunch of events.
As far as I understand this is covered by `test_deploy_ok_scenario` test.

I like the idea of having a common function to generate the "base scenario" (without playing it)... What about doing this:

```python
def generate_scenario():
    return Scenario.from_events(
        ("install", "config-changed", "start", Event("mimir-pebble-ready", workload=Mock()))
    )(MimirK8SOperatorCharm)
```

And in the tests:

```python
cc = generate_scenario().play_until_complete()
```

Reply:
I just pushed this change so you can see it.